### PR TITLE
Pbf 401

### PR DESCRIPTION
**Pull request by nagavenivn**

### CODE REVIEW

File: src/osmf/js/osmf_flash.js [hunk 1]
```diff
@@ -90,7 +90,7 @@
     }
     this.encodings = testForFlash();
     this.technology = "flash";
-    this.features = []; //when CC need to be enabled, it needs to be set to OO.VIDEO.FEATURE.CLOSED_CAPTIONS
+    this.features = ["closedCaptions"];
```
Comment (angieisrani) [inline src/osmf/js/osmf_flash.js]: Do you have html5-common loaded in this file?  I would recommend using OO.VIDEO.FEATURES.CLOSED_CAPTIONS

File: src/osmf/js/osmf_flash.js [hunk 2]
```diff
@@ -280,6 +280,31 @@
...
+     * @public
+     * @method OoyalaFlashVideoWrapper#setClosedCaptions
+     * @param {string} language Selected language of captions
+     * @param {object} closedCaptions The captions object
+     * @param {object} params The parameters object
+     */
+      
+     this.setClosedCaptions = function(language,closedCaptions,params){
+       var parameters = {language:language, closedCaptions:closedCaptions, params:params};
+        this.callToFlash("setVideoClosedCaptions()" , parameters);
```
Comment (angieisrani) [inline src/osmf/js/osmf_flash.js]: minor - there is some inconsistent spacing here.  I suggest spaces between input parameters to the function and reducing one leading space on this line

File: src/osmf/js/osmf_flash.js [hunk 3]
```diff
@@ -383,15 +408,15 @@
...
       // Reset the source
       this.setVideoUrl('');
 
       // Unsubscribe all events
+      
       this.unsubscribeAllEvents();
 
       // Pass destroy to flash plugin.
-      this.callToFlash("destroy");
+      this.callToFlash("destroy",null);
```
Comment (angieisrani) [inline src/osmf/js/osmf_flash.js]: if exactly one parameter is always expected, I would recommend having callToFlash turn undefined into null

File: src/osmf/lib/HDSPlayer.as [hunk 4]
```diff
@@ -208,10 +282,14 @@ package
      */
     private function onPlayComplete(event:TimeEvent):void
     {
+      if (_captionLabel.visible)
+      {
+        _captionLabel.visible = false;
+      }
```
Comment (angieisrani) [inline src/osmf/lib/HDSPlayer.as]: Out of curiosity - on a reply does this become true again?

File: src/osmf/lib/HDSPlayer.as [hunk 5]
```diff
@@ -419,7 +493,197 @@ package
...
+    /**
+     * Sets the closed captions mode
+     * @public
+     * @method HDSPlayer#onSetVideoClosedCaptionsMode
+     * @param {Event} event The event passed from the external interface.
+     */
+    public function onSetVideoClosedCaptionsMode(event:DynamicEvent):void
+    {
+      _mode = (String)(event.args);
+      _captionLabel.visible = false;
```
Comment (angieisrani) [inline src/osmf/lib/HDSPlayer.as]: why is this setting false instead of _mode?

File: utils/ClosedCaption/CaptioningLoader.as [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+					httpLoader.removeEventListener(LoaderEvent.LOAD_STATE_CHANGE, onHTTPLoaderStateChange);
+					updateLoadTrait(loadTrait, event.newState);
+				}
+			}
+			
+			function onLoadError(event:MediaErrorEvent):void
+			{
+				// Only remove this listener, as there will be a corresponding
+				// event for the load failure.
+				httpLoadTrait.removeEventListener(MediaErrorEvent.MEDIA_ERROR, onLoadError);
```
Comment (angieisrani) [inline utils/ClosedCaption/CaptioningLoader.as]: Is it a problem that event listener LOAD_STATE_CHANGE remains in this case?  What if a new cc is loaded, you might end up with two listeners on LOAD_STATE_CHANGE (and vica versa)

File: utils/ClosedCaption/CaptioningProxyElement.as [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+						}
+					}
+					else
+					{		
+						var timedTextURL:String = metadata.getValue(CaptioningPluginInfo.CAPTIONING_METADATA_KEY_URI);
+						if (timedTextURL != null)
+						{
+							_loadTrait = new CaptioningLoadTrait(new CaptioningLoader(), new URLResource(timedTextURL));
+							
+							_loadTrait.addEventListener(LoadEvent.LOAD_STATE_CHANGE, onLoadStateChange, false, 99);
```
Comment (angieisrani) [inline utils/ClosedCaption/CaptioningProxyElement.as]: what removes this listener?

File: src/osmf/js/osmf_flash.js (hunk 1 above)
Comment (nagavenivn) [inline src/osmf/js/osmf_flash.js]: We have hard coded this value as html5-commom is not loaded in the file. This will be resolved when PBF-471 is fixed.

File: src/osmf/lib/HDSPlayer.as (hunk 4 above)
Comment (angieisrani) [inline src/osmf/lib/HDSPlayer.as]: quick question - why was this removed?

Comment (nagavenivn) [inline src/osmf/lib/HDSPlayer.as]: Visibility of captions text box is set again on replay. Hence it is removed from here.

**Comment by angieisrani** [comment]
Some of the comments I made were against a library that was imported.  The changes are not going to be made against these files
